app/form_objects/result_filters/location_filter_form.rb

```diff
@@ -0,0 +1,82 @@
+module ResultFilters
+  class LocationFilterForm
+    NO_OPTION = nil
+    LOCATION_OPTION = "1".freeze
+
+    attr_reader :params, :errors
+
+    def initialize(params)
+      @params = params
+      @errors = nil
+      @valid = nil
+      validate
+    end
+
+    def valid?
+      @valid
+    end
+
+  private
+
+    def validate
+      case selected_option
+      when NO_OPTION
+        handle_no_option
+      when LOCATION_OPTION
+        if location_query.nil?
+          handle_missing_location
+        else
+          handle_location_option
+        end
+      else
+        @valid = true
+      end
+    end
+
+    def handle_location_option
+      geocode_params = geocode_params_for(location_query)
+      if geocode_params
+        @params.merge!(geocode_params)
+        @valid = true
+      else
+        @errors = "We couldn't find this location, please check your input and try again"
+        @valid = false
+      end
+    end
+
+    def handle_missing_location
+      @errors = "Please enter a postcode, town or city in England"
+      @valid = false
+    end
+
+    def handle_no_option
+      @errors = "Please choose an option"
+      @valid = false
+    end
+
+    def geocode_params_for(query)
+      results = Geocoder.search(query, components: "country:UK")
```

Comment:
@gpeng - I have not extracted the geocoding to a separate service as it's only one line. Don't see much point unless you think otherwise ...

Reply:
I was just thinking of not accessing the external dependency directly. Happy to leave it as is for now and move it when we use it somewhere else.